pkg/operator/operator.go

```diff
@@ -247,6 +248,19 @@ func (o *Operator) Start(ctx context.Context) error {
 	}
 }
 
+func (o *Operator) determineReplicasForDefaultIngressController() (int32, error) {
+	infraConfig := &configv1.Infrastructure{}
+	if err := o.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+		return 0, fmt.Errorf("failed fetching infrastructure config: %v", err)
+	}
+	ingressConfig := &configv1.Ingress{}
+	if err := o.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, ingressConfig); err != nil {
+		return 0, fmt.Errorf("failed fetching ingress config: %v", err)
```

Comment:
We're transitioning towards using wrapped errors in new code:
```suggestion
		return 0, fmt.Errorf("failed fetching infrastructure config: %w", err)
	}
	ingressConfig := &configv1.Ingress{}
	if err := o.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, ingressConfig); err != nil {
		return 0, fmt.Errorf("failed fetching ingress config: %w", err)
```